crates/miden-testing/tests/auth/multisig.rs

```diff
@@ -0,0 +1,400 @@
+use miden_lib::{
+    account::wallets::BasicWallet, note::create_p2id_note, transaction::TransactionKernel,
+};
+use miden_objects::{
+    Felt, Hasher, Word,
+    account::{Account, AccountBuilder, AccountStorageMode, AccountType, AuthSecretKey},
+    asset::FungibleAsset,
+    crypto::dsa::rpo_falcon512::{PublicKey, SecretKey},
+    note::NoteType,
+    testing::account_id::{
+        ACCOUNT_ID_REGULAR_PUBLIC_ACCOUNT_IMMUTABLE_CODE,
+        ACCOUNT_ID_REGULAR_PUBLIC_ACCOUNT_UPDATABLE_CODE,
+    },
+    transaction::{OutputNote, TransactionScript},
+    vm::AdviceMap,
+};
+use miden_testing::{Auth, MockChainBuilder};
+use miden_tx::{
+    TransactionExecutorError,
+    auth::{BasicAuthenticator, SigningInputs, TransactionAuthenticator},
+    utils::word_to_masm_push_string,
+};
+use rand::SeedableRng;
+use rand_chacha::ChaCha20Rng;
+use vm_processor::crypto::RpoRandomCoin;
+
+// ================================================================================================
+// HELPER FUNCTIONS
+// ================================================================================================
+
+type MultisigTestSetup = (Vec<SecretKey>, Vec<PublicKey>, Vec<BasicAuthenticator<ChaCha20Rng>>);
+
+/// Sets up secret keys, public keys, and authenticators for multisig testing
+fn setup_keys_and_authenticators(
+    num_approvers: usize,
+    num_signers: usize, // How many of the approvers will actually sign
+    seed: [u8; 32],
+) -> anyhow::Result<MultisigTestSetup> {
+    let mut rng = ChaCha20Rng::from_seed(seed);
+
+    let mut secret_keys = Vec::new();
+    let mut public_keys = Vec::new();
+    let mut authenticators = Vec::new();
+
+    for _ in 0..num_approvers {
+        let sec_key = SecretKey::with_rng(&mut rng);
+        let pub_key = sec_key.public_key();
+
+        secret_keys.push(sec_key);
+        public_keys.push(pub_key);
+    }
+
+    // Create authenticators only for the signers we'll actually use
+    for i in 0..num_signers {
+        let authenticator = BasicAuthenticator::<ChaCha20Rng>::new_with_rng(
+            &[(public_keys[i].into(), AuthSecretKey::RpoFalcon512(secret_keys[i].clone()))],
+            rng.clone(),
+        );
+        authenticators.push(authenticator);
+    }
+
+    Ok((secret_keys, public_keys, authenticators))
+}
+
+/// Creates a multisig account with the specified configuration
+fn create_multisig_account(
+    threshold: u32,
+    public_keys: &[PublicKey],
+    account_seed: [u8; 32],
```

Comment:
Since this is only ever used as a helper, and the `account_seed` is the same for all, let's hardcode that instead of passing it as a parameter